### PR TITLE
Make demo titles and indexing more consistent

### PR DESCRIPTION
**Pull request by huonw**
This goes through all our demos and adjusts the titles to be a bit more consistent and punchy, e.g. more consistent capitalisation, no mentions of "StellarGraph" or "example", no mentions of datasets (a user doesn't care about datasets).

These new titles are most visible in:
- the demo table of contents: https://stellargraph--1474.org.readthedocs.build/en/1474/demos/index.html#table-of-contents
- READMEs like:
  - embeddings: https://github.com/stellargraph/stellargraph/blob/ff70bb295dd4e29f2449a2c969d2e06d971f11dc/demos/embeddings/README.md
  - link prediction: https://github.com/stellargraph/stellargraph/blob/ff70bb295dd4e29f2449a2c969d2e06d971f11dc/demos/link-prediction/README.md
  - node classification: https://github.com/stellargraph/stellargraph/blob/ff70bb295dd4e29f2449a2c969d2e06d971f11dc/demos/node-classification/README.md

See: #1473

### CODE REVIEW

File: demos/connector/neo4j/directed-graphsage-on-cora-neo4j-example.ipynb [hunk 1]
```diff
@@ -4,7 +4,7 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "# Stellargraph example: directed CORA citation network with directed GraphSAGE and Neo4J.\n"
+    "# Directed GraphSAGE with Neo4j, on Cora\n"
```
Comment (timpitman) [inline demos/connector/neo4j/directed-graphsage-on-cora-neo4j-example.ipynb:7]:
```suggestion
    "# Directed GraphSAGE with Neo4J, on Cora\n"
```

File: demos/connector/neo4j/undirected-graphsage-on-cora-neo4j-example.ipynb [hunk 2]
```diff
@@ -4,7 +4,7 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "# Undirected GraphSAGE on the CORA citation network using StellarGraph and Neo4j"
+    "# GraphSAGE with Neo4j, on Cora"
```
Comment (timpitman) [inline demos/connector/neo4j/undirected-graphsage-on-cora-neo4j-example.ipynb:7]:
```suggestion
    "# GraphSAGE with Neo4J, on Cora"
```

**Comment by timpitman** [comment]
I think this PR misses a typo in "Detection of Twitter users who use hateful lexicon using graph machine learning with Stellargraph".  (StellarGraph)?

**Comment by huonw** [comment]
That's true: I initially didn't do the community detection or either of the hateful twitter examples (#1477 is related), but I guess I should...

File: demos/connector/neo4j/directed-graphsage-on-cora-neo4j-example.ipynb (hunk 1 above)
Comment (huonw) [reply demos/connector/neo4j/directed-graphsage-on-cora-neo4j-example.ipynb]: Hm, actually their website uses `Neo4j`, e.g. https://neo4j.com

> What is Neo4j?

Comment (timpitman) [reply demos/connector/neo4j/directed-graphsage-on-cora-neo4j-example.ipynb]: well spotted 😊

**Comment by huonw** [comment]
As subfolder titles, those will disappear with #1471, but I'll remove them here just in case #1471 doesn't land.

**Pull request merged**